File: LeetCode/102BinaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        # BFS -> Time:O(n), Space:O(2^h)
        ans = []
        if not root:
            return ans
        q = []
        q.append(root)
        while q:
            level = []
            size = len(q)
            for i in range(size):
                front = q.pop(0)
                if front.left:
                    q.append(front.left)
                if front.right:
                    q.append(front.right)
                level.append(front.val)
            ans.append(level)
        return ans

File: LeetCode/102BinaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        // BFS -> Time:O(n), Space:O(2^h)
        vector<vector<int>> ans;
        if (root == NULL) {
            return ans;
        }
        int size;
        TreeNode *front;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()) {
            vector<int> level;
            size = q.size();
            for(int i = 0; i < size; i++) {
                front = q.front();
                q.pop();
                level.push_back(front->val);
                if (front->left) {
                    q.push(front->left);
                }
                if (front->right) {
                    q.push(front->right);
                }
            }
            ans.push_back(level);
        }
        return ans;
    }
};
